circuit_maintenance_parser/parser.py

```diff
@@ -37,10 +37,25 @@ def get_data_types(cls) -> List[str]:
         """Return the expected data type."""
         return cls._data_types
 
-    def parse(self, raw: bytes) -> List[Dict]:
-        """Extract a list of data that partially or completely describes a series of Maintenance objects."""
+    def parser_hook(self, raw: bytes):
+        """Custom parser logic."""
         raise NotImplementedError
 
+    def parse(self, raw: bytes) -> List[Dict]:
+        """Execute parsing."""
+        try:
+            result = self.parser_hook(raw)
+            if result == [{}]:
+                raise ParserError(
+                    f"{self.__class__.__name__} has not extracted any valid data has been extracted from notification:"  # type: ignore
+                    f"\n{raw}"  # type: ignore
+                )
+            logger.debug("Successful parsing for %s", self.__class__.__name__)
+            return result
+
+        except Exception as exc:
+            raise ParserError from exc
```

Comment:
Note that the inner ParserError will be caught by the outer `try...except` block, which will wrap it in another ParserError... Perhaps instead?

```suggestion
        try:
            result = self.parser_hook(raw)
        except Exception as exc:
            raise ParserError from exc
        if result == [{}]:
            raise ParserError(
                f"{self.__class__.__name__} has not extracted any valid data has been extracted from notification:"  # type: ignore
                f"\n{raw}"  # type: ignore
            )
        logger.debug("Successful parsing for %s", self.__class__.__name__)
        return result
```